### DISCUSSION TITLE
Does whereRelation supports IN query?

### DISCUSSION THREAD

**Discussion by HilalLko** [laravel/framework#43912] [Q&A] [answered]
I have a condition where I need to filter users based on multiple values in the associated relation.
For example:
My User Model Looks like
```
class User extends Model
{
    public function role()
    {
        return $this->belongsTo(Role::class);
    }
```
and my Role model looks like
```
class Role extends Model
{
    public function users()
    {
        return $this->hasMany(User::class);
    }
```

when filtering users with a single role 
```
User::whereRelation('role','name','student')->get();
```
am getting desired results.

I just wanted to know, can I utilize The whereIn method to verify that a given relation column's value is contained within the given array:

```
User::whereRelationIn('role','name',['student','parent','teacher'])->get();
```

**Comment by erikn69** [comment]
Use `whereHas` with a closure and a `whereIn`
```php
User::whereHas('role', fn($q) => $q->whereIn('name',['student','parent','teacher']))->get();
```

**Reply by HilalLko** [reply]
Well @erikn69 `whereHas` with `closure` is already implemented, I was specifically looking for something new which utilizes updated `whereRelation`

**Reply by erikn69** [reply]
So do a PR, and that would also mean support for `whereMonth, whereYear, whereDay, whereBetween, whereColumn and others` including the `whereNot` of them

**Reply by HilalLko** [reply]
@erikn69, No need to provide additional PR as this [PR](https://github.com/laravel/framework/pull/42491) does what is required and even supports your suggested `whereMonth, whereYear, whereDay, whereBetween, whereColumn`

as with mentioned PR we can have closure with `whereRelation`
```
User::whereRelation('role', fn($q) => $q->whereIn('name',['student','parent','teacher']))->get();
```

**Comment by HilalLko** [comment, marked as answer]
As this [PR](https://github.com/laravel/framework/pull/42491) does what is required and even supports suggested `whereMonth, whereYear, whereDay, whereBetween, whereColumn`

With mentioned PR we can have closure with `whereRelation`
```
User::whereRelation('role', fn($q) => $q->whereIn('name',['student','parent','teacher']))->get();
```
Generated query with closure base whereRelation
```
select * from `users` where exists (select * from `roles` where `users`.`role_id` = `roles`.`id` and `name` in ('student','parent','teacher')) and `users`.`deleted_at` is null
```

**Reply by siarheipashkevich** [reply]
in that case we can just use `whereHas` method instead of `whereRelation`

**Reply by HilalLko** [reply]
Well @siarheipashkevich You can prefer any of either, but to keep query more readable and clean we can simply use `whereRelation` if we don't have complex condition.
Indeed `whereHas` is more powerful & flexible and can be a good choice if Laravel version being used doesn't support `whereRelation`
